src/Microsoft.Framework.BufferedHtmlContent.Sources/BufferedHtmlContent.cs

```diff
@@ -147,5 +147,16 @@ public override string ToString()
                 return writer.ToString();
             }
         }
+
+        // Avoid encoding newline characters
+        private class NewLine : IHtmlContent
```

Comment:
Suggest this class should implement `AppendEncoded(string)` using another `private` class. We're going that route (i.e. that method should be present in `IHtmlContentBuilder`) and the value of moving `HtmlString` some place central seems very low.